api/tasks.go

```diff
@@ -282,16 +285,6 @@ func NewDefaultReschedulePolicy(jobType string) *ReschedulePolicy {
 			Unlimited: pointerOf(false),
 		}
 
-	case "system":
-		dp = &ReschedulePolicy{
-			Attempts:      pointerOf(0),
-			Interval:      pointerOf(time.Duration(0)),
-			Delay:         pointerOf(time.Duration(0)),
-			DelayFunction: pointerOf(""),
-			MaxDelay:      pointerOf(time.Duration(0)),
-			Unlimited:     pointerOf(false),
-		}
```

Comment:
Question for reviewers: arguably we should be returning `nil` here for system and sysbatch jobs, and then catching that in `Canonicalize`. I've fixed `Canonicalize` but I was hesitant to change the value returned here because this method is part of the public API. But it's never been correct to send the initialized `ReschedulePolicy`, so if you're relying on this you're getting bad results already today. Should we just change the output so that it returns nil and erase it when we call `Canonicalize` in case a user has set it on their own?

Reply:
I wonder if we could not touch it within the `Canonicalize` function and validate that it is empty within our inner `Validate` logic?

My thinking is that if a job specification author does not realise `sysbatch` and `system` jobs don't support `Reschedule` blocks and we silently nil it out, the job will deploy without indication of the change and they might assume reschedule is in place.

Reply:
> I wonder if we could not touch it within the `Canonicalize` function and validate that it is empty within our inner `Validate` logic?

The fix I did to `Canonicalize` leaves anything the user set untouched for system/sysbatch, but are you suggesting we validate that it's nil _or_ the empty/unset struct with all zero values? That would accept a wider range of values (including this default we're removing) but avoid the expectation that it does anything.